### DISCUSSION TITLE
Frequent socket disconnetion with `ping timeout` in client side with stable internet connection

### DISCUSSION THREAD

**Discussion by itzmanish** [socketio/socket.io#4362] [General]
Hi, So I am facing some issues with socket io (v4.4.1) and socket.io-client(v4.2.0). That is some clients get disconnected with the reason `ping timeout` on the client but the server gets the reason `transport close`. Now here I can verify that server didn't have any high load and the client has a pretty good and stable internet connection. 

There were other clients connected to the same socket server and they didn't face any disconnection so I can say that server was sending `ping` packets properly to all the clients. 

Any help on getting the root cause of this issue will be very helpful and highly appreciated.

Thank you.

**Comment by darrachequesne** [comment]
Hi!

This might happen if the WebSocket connection is severed but not completely closed. The server sends a ping packet, which fails (hence the "transport close" disconnection), and the client eventually gets a timeout (without knowing the WebSocket connection is broken).

Are you able to reproduce the issue? Is it linked to a given browser?

**Reply by itzmanish** [reply]
I am not able to reproduce `transport close` error but If I manually produce an error on ws.write() I get `transport error`.
